activejob/lib/active_job/queue_name.rb

```diff
@@ -29,22 +29,35 @@ def queue_name_from_part(part_name) #:nodoc:
         name_parts = [queue_name_prefix.presence, queue_name]
         name_parts.compact.join(queue_name_delimiter)
       end
+
+      def queue_name=(v)
+        @queue_name = v
+      end
+
+      def queue_name
+        name = defined?(@queue_name) ? instance_variable_get(:@queue_name) : default_queue_name
```

Comment:
Also, unless you specifically want to handle user-defined `nil` or `false` for `@queue_name`, you can just do:

def queue_name
  @queue_name || default_queue_name
end